go/vt/vtgate/vindexes/lookup_internal.go

```diff
@@ -178,13 +202,9 @@ func (lkp *lookupInternal) createCustom(vcursor VCursor, rowsColValues [][]sqlty
 	}
 	fmt.Fprintf(buf, "%s) values(", lkp.To)
 
-	bindVars := make(map[string]*querypb.BindVariable, 2*len(rowsColValues))
-	// Make a copy before sorting.
-	rowsColValues = append([][]sqltypes.Value(nil), rowsColValues...)
-	toValues = append([]sqltypes.Value(nil), toValues...)
```

Comment:
The subtle line, which is easy to miss is the `sort` in the next line. This is the reason why we have to create a copy because we'll break the caller if we changed the input parameters (and we did).

Since a copy was needed anyways, I create a filtered "copy" now instead of blind copying.